pkg/alizer/alizer.go

```diff
@@ -47,3 +52,70 @@ func GetDevfileLocationFromDetection(typ recognizer.DevFileType, registry api.Re
 		DevfileRegistry: registry.Name,
 	}
 }
+
+// DetectName retrieves the name of the project (if available)
+// If source code is detected:
+// 1. Detect the name (pom.xml for java, package.json for nodejs, etc.)
+// 2. If unable to detect the name, use the directory name
+//
+// If no source is detected:
+// 1. Use the directory name
+//
+// Last step. Sanitize the name so it's valid for a component name
+
+// Use:
+// import "github.com/redhat-developer/alizer/pkg/apis/recognizer"
+// components, err := recognizer.DetectComponents("./")
+
+// In order to detect the name, the name will first try to find out the name based on the program (pom.xml, etc.) but then if not, it will use the dir name.
+func DetectName(path string) (string, error) {
+	if path == "" {
+		return "", fmt.Errorf("path is empty")
+	}
+
+	// Check if the path exists using os.Stat
+	dir, err := os.Stat(path)
+	if err != nil {
+		return "", err
+	}
+
+	// Check to see if the path is a directory, and fail if it is not
+	if !dir.IsDir() {
+		return "", fmt.Errorf("alizer DetectName %q path is not a directory", path)
+	}
+
+	// Step 1.
+	// Get the name of the directory from the devfile absolute path
+	// Use that path with Alizer to get the name of the project,
+	// if unable to find the name, we will use the directory name
+	components, err := recognizer.DetectComponents(path)
+	if err != nil {
+		return "", err
+	}
+	klog.V(4).Infof("Found components: %v", components)
+
+	// Take the first name that is found
+	var detectedName string
+	if len(components) > 0 {
+		detectedName = components[0].Name
+	}
+
+	// Step 2. If unable to detect the name, we will use the directory name.
+	// Alizer will not correctly default to the directory name when unable to detect it via pom.xml, package.json, etc.
+	// So we must do it ourselves
+	// The directory name SHOULD be the path (we use a previous check to see if it's "itsdir"
+	if detectedName == "" {
+		detectedName = path
```

Comment:
Now it returns `home-pvala-something` for /home/pvala/something directory, don't we want the name to just be `something`?